src/platform_linux/platform_inline.h

```diff
@@ -453,8 +459,17 @@ platform_aligned_malloc(const platform_heap_id heap_id,
    const size_t padding  = platform_alignment(alignment, size);
    const size_t required = (size + padding);
 
-   void *retptr = (heap_id ? splinter_shm_alloc(heap_id, required, objname)
-                           : aligned_alloc(alignment, required));
+   void *retptr = NULL;
+   if (heap_id == PROCESS_PRIVATE_HEAP_ID) {
```

Comment:
Clarified usage of the semantic of `heap_id`; NULL means process-private-heap-ID, so go thru old `malloc()`-style code-flow.